tests/distributions/test_reparameterize.py

```diff
@@ -0,0 +1,107 @@
+import pytest
+import torch
+from torch.autograd import grad
+
+import pyro
+import pyro.distributions as dist
+from pyro import poutine
+from pyro.distributions.reparameterize import LocScaleReparameterizer, TrivialReparameterizer
+from pyro.distributions.stable import StableReparameterizer
+from tests.common import assert_close
+
+
+# Test helper to extract a few central moments from samples.
+def normal_probe(x):
+    m1 = x.mean(0)
+    x = x - m1
+    xx = x * x
+    xxx = x * xx
+    xxxx = xx * xx
+    m2 = xx.mean(0)
+    m3 = xxx.mean(0) / m2 ** 1.5
+    m4 = xxxx.mean(0) / m2 ** 2
+    return torch.stack([m1, m2, m3, m4])
+
+
+@pytest.mark.parametrize("shape", [(), (4,), (3, 2)])
+@pytest.mark.parametrize("Reparam", [
+    TrivialReparameterizer,
+    LocScaleReparameterizer,
+])
+def test_normal(shape, Reparam):
+    loc = torch.empty(shape).uniform_(-1., 1.).requires_grad_()
+    scale = torch.empty(shape).uniform_(0.5, 1.5).requires_grad_()
+
+    def model():
+        with pyro.plate_stack("plates", shape):
+            with pyro.plate("particles", 100000):
+                pyro.sample("x", dist.Normal(loc, scale))
+
+    value = poutine.trace(model).get_trace().nodes["x"]["value"]
+    expected_probe = normal_probe(value)
+
+    def config_fn(site):
+        if site["name"] == "x" and isinstance(site["fn"], dist.Normal):
+            return {"reparam": Reparam()}
+        return {}
```

Comment:
I guess we don't need to return an empty dict here?

Reply:
A dict is required by `infer_config`'s `config_fn`